query-engine/query-engine/src/logger.rs

```diff
@@ -1,43 +1,47 @@
 use opentelemetry::{
-    global,
-    sdk::{
-        propagation::TraceContextPropagator,
-        trace::{Config, Tracer},
-        Resource,
-    },
+    sdk::{trace::Config, Resource},
     KeyValue,
 };
 use opentelemetry_otlp::WithExportConfig;
-use query_core::is_user_facing_trace_filter;
+use query_core::telemetry;
 use query_engine_metrics::MetricRegistry;
 use tracing::{dispatcher::SetGlobalDefaultError, subscriber};
-use tracing_subscriber::{filter::filter_fn, layer::SubscriberExt, EnvFilter, Layer};
+use tracing_subscriber::{filter::filter_fn, layer::SubscriberExt, Layer};
 
 use crate::LogFormat;
 
 type LoggerResult<T> = Result<T, SetGlobalDefaultError>;
 
 /// An installer for a global logger.
 #[derive(Debug, Clone)]
-pub struct Logger<'a> {
+pub struct Logger {
     service_name: &'static str,
     log_format: LogFormat,
-    enable_telemetry: bool,
     log_queries: bool,
-    telemetry_endpoint: Option<&'a str>,
+    tracing_config: TracingConfig,
     metrics: Option<MetricRegistry>,
 }
 
-impl<'a> Logger<'a> {
+// TracingConfig specifies how tracing will be exposed by the logger facility
+#[derive(Debug, Clone)]
+enum TracingConfig {
+    // exposed means tracing will be exposed through an HTTP endpoint in a jaeger-compatible format
+    Exposed(String),
```

Comment:
Is the string data the endpoint?

Owned strings look easier on the surface but they open other challenges with the borrow checker, and often harder ones — was it made owned for convenience?

Reply:
Yes, it only was made owned for convenience. I can introduce a lifetime, and see how it looks like.